### PR TITLE
Schedule Microphone Jobs

### PR DESCRIPTION
**Pull request by orkosinha**
Microphone is currently a `volume-backend` duplicate when it comes to refreshing articles.

This PR adds scheduled tasks and a README.

Testing was done through modifying `volume-backend` to use Microphone collections and checking functionality.

## Next Steps
* Add logging
* Deployment

### CODE REVIEW

File: .envrctemplate [hunk 1]
```diff
@@ -1,17 +1,21 @@
-export NODE_ENV=[node env]
-export APP_PORT=[port]
-export MONGO_ADDRESS=[db uri]
+# create states folder
```
Comment (alanna-zhou) [inline .envrctemplate]: delete comment

File: .envrctemplate [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
     echo "Activating $(python -V) virtualenv"
-    source .venv/bin/activate
+    source venv/bin/activate
+
+    # announce python version and show the path of the current python in ${PATH}
+    echo "Virtualenv has been activated for $(python -V)"
+    echo "Python is running from $(which python)"
+else
+    echo "Requested Python version not installed with pyenv"
 fi
```
Comment (alanna-zhou) [inline .envrctemplate:17]: damn this is intense

File: src/app.py [hunk 3]
```diff
@@ -1,26 +1,51 @@
 import json
+import logging
 import os
+import schedule
+import time
 from article import Article
 from publication import Publication
 from pymongo import MongoClient, UpdateOne
 
+# Open publications json file
```
Comment (alanna-zhou) [inline src/app.py]: delete

File: src/app.py [hunk 4]
```diff
@@ -1,26 +1,51 @@
...
+    UpdateOne({"_id": p["_id"]}, {"$setOnInsert": p}, upsert=True) for p in publications
+]
+# Add publications to db
+with MongoClient(MONGO_ADDRESS) as client:
+    db = client.microphone
+    result = db.publications.bulk_write(publication_upserts)
 
-MONGO_ADDRESS = os.getenv("MONGO_ADDRESS")
-client = MongoClient(MONGO_ADDRESS)
+# Fucntion for gathering articles for running with scheduler
```
Comment (alanna-zhou) [inline src/app.py]: spelling error

File: src/app.py [hunk 5]
```diff
@@ -1,26 +1,51 @@
...
+def gather_articles():
+    articles = []
+    for publication in publications_json:
+        p = Publication(publication)
+        for entry in p.get_feed():
+            articles.append(Article(entry, publication["slug"]).serialize())
+
+    article_upserts = [
```
Comment (alanna-zhou) [inline src/app.py:33]: what is an upsert haha

**Review by alanna-zhou — APPROVED** [review]
just some tiny comments

File: src/app.py (hunk 5 above)
Comment (yuna878) [reply src/app.py:33]: it's insert + update lol 
Insert if the object is a new object // update if object already exists

**Review by yuna878 — APPROVED** [review]
LGTM

File: src/app.py (hunk 5 above)
Comment (alanna-zhou) [reply src/app.py:33]: LOOOLLL damnnnnnn how did i not know that 😂  thankuu

File: .envrctemplate [hunk 6]
```diff
@@ -1,17 +1,21 @@
-export NODE_ENV=[node env]
-export APP_PORT=[port]
-export MONGO_ADDRESS=[db uri]
+# create states folder
+mkdir ./states
 
 # check if python version is set in current dir
```
Comment (tedim52) [inline .envrctemplate:3]: i've never seen this before, curious, is this to ensure that the correct python installation is being used when working?

File: README.md [hunk 7]
```diff
@@ -1 +1,26 @@
-# volume-microservice
\ No newline at end of file
+# volume-microservice
+
+[![Python 3.9.2](https://img.shields.io/badge/python-3.9.2-blue.svg)](https://www.python.org/downloads/release/python-392/)
+
+A project by [Cornell AppDev](http://cornellappdev.com), a project team at Cornell University.
```
Comment (tedim52) [inline README.md:5]: Can we add a description of its relation to volume-backend?

File: src/app.py [hunk 8]
```diff
@@ -1,26 +1,51 @@
...
-    publications.append(pub.serialize())
+MONGO_ADDRESS = os.getenv("MONGO_ADDRESS")
 
-publication_upserts = [ UpdateOne({'_id':p['_id']}, {'$setOnInsert':p}, upsert=True) for p in publications ]
-article_upserts = [ UpdateOne({'articleUrl':a['articleUrl']}, {'$setOnInsert':a}, upsert=True) for a in articles ]
+# Get serialized publications
+publications = [Publication(p).serialize() for p in publications_json]
+publication_upserts = [
+    UpdateOne({"_id": p["_id"]}, {"$setOnInsert": p}, upsert=True) for p in publications
+]
```
Comment (tedim52) [inline src/app.py:19]: i take it back... Orko code is excellent documentation

**Review by tedim52 — APPROVED** [review]
minor comments! but LGTM

File: .envrctemplate (hunk 6 above)
Comment (orkosinha) [reply .envrctemplate:3]: Yep, also automatically loads and unloads a virtual environment using direnv

File: src/app.py (hunk 4 above)
Comment (orkosinha) [reply src/app.py]: rip

File: README.md (hunk 7 above)
Comment (orkosinha) [reply README.md:5]: You got it

File: src/app.py (hunk 8 above)
Comment (orkosinha) [reply src/app.py:19]: list comprehensions are a bit excessive here lol

**Pull request merged**